src/main/java/seedu/address/storage/XmlAdaptedLoan.java

```diff
@@ -260,31 +155,32 @@ public Loan toModelType() throws IllegalValueException {
             loanTags.add(tag.toModelType());
         }
 
-        checkNameValid();
+        checkFieldValid(name, Name.class, Name::isValidName, Name.MESSAGE_NAME_CONSTRAINTS);
         final Name modelName = new Name(name);
 
-        checkNricValid();
+        checkFieldValid(nric, Nric.class, Nric::isValidNric, Nric.MESSAGE_NRIC_CONSTRAINTS);
         final Nric modelNric = new Nric(nric);
 
-        checkPhoneValid();
+        checkFieldValid(phone, Phone.class, Phone::isValidPhone, Phone.MESSAGE_PHONE_CONSTRAINTS);
         final Phone modelPhone = new Phone(phone);
 
-        checkEmailValid();
+        checkFieldValid(email, Email.class, Email::isValidEmail, Email.MESSAGE_EMAIL_CONSTRAINTS);
         final Email modelEmail = new Email(email);
 
-        checkAddressValid();
+        checkFieldValid(address, Address.class, Address::isValidAddress, Address.MESSAGE_ADDRESS_CONSTRAINTS);
         final Address modelAddress = new Address(address);
 
-        checkLoanStatusValid();
+        checkFieldValid(loanStatus, LoanStatus.class, LoanStatus::isValidLoanStatus,
+            LoanStatus.MESSAGE_LOANSTATUS_CONSTRAINTS);
         final LoanStatus modelLoanStatus = LoanStatus.valueOf(loanStatus);
 
-        checkBikeValid();
+        checkFieldValid(bike, Bike.class, Name::isValidName, Name.MESSAGE_NAME_CONSTRAINTS);
         final Bike modelBike = new Bike(new Name(bike));
 
-        checkLoanRateValid();
+        checkFieldValid(rate, LoanRate.class, LoanRate::isValidRate, LoanRate.MESSAGE_LOANRATE_CONSTRAINTS);
         final LoanRate modelRate = new LoanRate(rate);
 
-        checkLoanTimeValid();
+        checkFieldValid(time, LoanTime.class, LoanTime::isValidLoanTime, LoanTime.MESSAGE_LOANTIME_CONSTRAINTS);
```

Comment:
LoanTime is not supposed to be part of this check as it is nullable.
In fact, there will be 2 copies of LoanTime that is passed into this. Will fix in my PR.